src/base/castlerandom.pas

```diff
@@ -67,22 +68,50 @@ procedure TCastleRandom.Initialize(RandomSeed: LongWord);
 begin
   if RandomSeed = 0 then
   begin
+    Seed := GetRandomSeed;
+  end
+  else seed := LongInt(RandomSeed);
+end;
+
+{$IFDEF Linux}
+function DEV_URANDOM: longint;
+var dev_rnd: file of integer;
+begin
+  { algorithm according to http://wiki.freepascal.org/Dev_random
+   /dev/urandom is a native *nix very high-quality random number generator.
+   it's 1000 times slower than CastleRandom,
+   but provides a perfect seed initialization. }
+  //filemode := 0;
+  AssignFile(dev_rnd, '/dev/urandom');
+  reset(dev_rnd);
+  read(dev_rnd,result);
+  CloseFile(dev_rnd);
+end;
+{$ENDIF}
+
```

Comment:
I'm unsure about having a special implementation for Linux.

It feels like we're reinventing here something that we shouldn't. We should use the same approach to Randomize as the FPC sources. And if there is a problem in FPC implementation of Randomize, then we should fix the FPC too (especially since using CastleRandom is optional; most applications will use a simple Random, and it's Ok if they don't need the speedup of CastleRandom).

Also, Linux is not the Unix, so in general it's good to avoid {$ifdef LINUX}, and use {$ifdef UNIX}. Unless you really have something specific to the Linux kernel. Your implementation is probably OK for FreeBSD, Darwin (Mac OS X), Android too. (But it should be tested.)

Also, choosing an OS-specific solution often increases the risk of introducing some bugs and not catching them. E.g. I usually use Linux, so the other branch will remain untested by me:) That's not some definite problem, after all we have a lot of OS-specific code in the engine anyway. But when a cross-platform solution is possible, I would prefer it a little.

FPC has been around for a long time, and it's standard library is thoroughly tested on various OSes --- they did the necessary decision-making *and* testing for us:) So I would prefer here to avoid any $ifdefs, and have a simple cross-platform implementation that follows the same approach as FPC Randomize.